### PR TITLE
Disable fetch calls when offline

### PR DESCRIPTION
**Pull request by csduarte**
#### Summary
Set's Client4 online status,  when `this.online == false` avoid making fetch calls.
Redux actions will reconcile the error.

This is to avoid the `OutOfMemoryError` caused on Android.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-10773

### CODE REVIEW

**Comment by miguelespinoza** [comment]
@enahum and @hmhealey here's the fix for OOM error

File: src/client/client4.js [hunk 1]
```diff
@@ -2372,6 +2377,13 @@ export default class Client4 {
     // Client Helpers
 
     doFetch = async (url, options) => {
+        if (!this.online) {
```
Comment (hmhealey) [inline src/client/client4.js]: Shouldn't this be in `doFetchWithResponse`?

Comment (hmhealey) [reply src/client/client4.js]: I'd just expect this check to cover all fetch calls made by Client4. Having it not cover a few of them seems weird to me

Comment (miguelespinoza) [reply src/client/client4.js]: Sounds fair.  I'll test to make sure everything looks good moving to ` doFetchWithResponse`

**Comment by miguelespinoza** [comment]
@enahum and @hmhealey done 😄

**Comment by jwilander** [comment]
@enahum if this is for 5.0 we'll need to update the base branch

**Comment by enahum** [comment]
@jwilander this is for 1.9, anyway I could update the base branch to `release-5.0` and do the same for the RN app.

thoughts??

**Comment by jwilander** [comment]
Yeah let's do that 👍

**Pull request merged**